### DISCUSSION TITLE
Max Wait time setting

### DISCUSSION THREAD

**Discussion by rayw45** [kubilus1/autoortho#460] [General]
I have heard conflicting advice regarding what to best set the Max Wait time too in the AO 7.1 config file.  Some say keep it short like the default 0.5 seconds otherwise if the scenery tile has not downloaded then X Plane will likely crash.  Others say make it longer like 4s or 5s so there is time for the missing needed tile to download before X Plane 12 crashes with a missing scenery file, but conversely will this cause X Plane to crash if it  waits too long ???

I do get X Plane 12 crashing to desktop occasionally with the log report clearly showing it is an AO related problem. Here is the last lines of my log.txt from last nights flight from VOBL to VABB in India in the KA350:

2:36:31.713 E/TEX: Fatal error when loading: Custom Scenery/z_ao_asi/textures/29328_46160_BI16.dds (EXCEPTION_IN_PAGE_ERROR, NTSTATUS=0xC0000002)
--=={This application has crashed!}==--
(Art controls are modified.)
--=={FILE: G:\SteamLibrary\steamapps\common\X-Plane 12\Log_ATC.txt}==--
--=={UUID: 9adab953-4bfb-4a18-b92a-d4f727fbf662}==--

This is the same type .dds texture file I always see when I do get these occasional crashes to desktop. Frustratingly it always happens after an hour or two flying in areas that I have not flown before so the needed scenery tiles are not yet in my AO Cache folder. You will see in the above case this happened after 2 hours 36 minutes into my flight which was nearing my destination. For the previous over 2 hours I had been flying at more than 300 Kts GSP over never previously flown territory too without any problem or any noticeable pauses either :-) 

My Max Wait time was set at 3.5s which I had increased from 2.5s before this flight after a previous same error type crash.  So which is the best way to reduce the chances of this happening again, reduce the max wait time or increase it say to 5s ???  I am sure many others here, including myself, would also be interested to know from the developers who will be in the know of course.

Thanks in advance for any help here and keep up the good work on this fantastic project for X Plane 12

**Comment by rayw45** [comment]
Nobody ever answered this question which surely one of the developers should know the answer.

It has got annoying when I fly a new flight and say an hour into the flight suddenly it crashed XP12 back to desktop when it is trying to download new tiles not already in my cache.  I am currently flying the Dash 8 Q400 which is a mid speed aircraft cruising at between 300 to 350Kts GS. I run in Windows 10 and Dokan. I have tried WinFSP and makes no difference. 

So once again which way is it best to adjust the max wit time to avoid this happening?  Currently I am set to 4 seconds.   Is it a longer wait to give it chance to download, or does the longer max wait cause X Plane to give up and crash?  Please can somebody help me as AO 0.7.1 works very well apart from having to still set debug = True

**Reply by TinkerNZ** [reply]
Hey Ray,
Unfortunately replies here are few and far between. I guess especially this time of the year.
Regards wait times, anything between 2.0 to 5.0 should aid the download problem. If the problem persists I suggest there may be other factors which slow down your particular connection.
Possible things to check:
- have you excluded AO from your anti virus checking? this would slow the many many cache files getting into your system.
- is your IP address being throttled by the source? Try monitoring your receive rate with Task Manager to see if it is steady and consistent. If you suspect very slow receiving try using a VPN connection which would give you a completely different IP address and see if things improve.
- are you running other apps which might be taking bandwidth priority?
Hope this may help, but it is a trial and error excercise on a per setup basis. What works for me may not work for you.
Cheers

**Reply by rayw45** [reply]
Thanks Tinker.  I have replied to you in the X Plane Org forum already.  Would be nice to get some response from the developers too as they must know answers to many of these questions.
